pkg/gitops/apply.go

```diff
@@ -0,0 +1,56 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+// GitOps can set up a repo as a gitops repo with flux
+type GitOps struct {
+	ClusterConfig    *api.ClusterConfig
+	UsersRepoOpts    git.Options
+	QuickstartName   string
+	FluxInstaller    *flux.Installer
+	ProfileGenerator *Profile
+	GitClient        *git.Client
+}
+
+// Apply setups gitops in a repository and a cluster and installs flux, helm, tiller and a quickstart into the cluster
+func (g *GitOps) Apply(ctx context.Context) error {
+
+	// Install Flux, Helm and Tiller. Clones the user's repo
+	err := g.FluxInstaller.Run(context.Background())
```

Comment:
I think we should move the cloning/pushing out of the flux installer, so that it doesn't own it (it's a shared resource by  both of them).

Among other things, it will allow for having a single add, commit and push.

Reply:
Good idea. To be done in a different PR